specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/ipAllocation.json

```diff
@@ -0,0 +1,395 @@
+      "get": {
+        "tags": [
+          "IpAllocations"
+        ],
+        "operationId": "IpAllocations_ListByResourceGroup",
+        "description": "Gets all IpAllocations in a resource group.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns a list of IpAllocation resources.",
+            "schema": {
+              "$ref": "#/definitions/IpAllocationListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List IpAllocations in resource group": {
+            "$ref": "./examples/IpAllocationListByResourceGroup.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "IpAllocationPropertiesFormat": {
+      "properties": {
+        "subnet": {
+          "$ref": "./virtualNetwork.json#/definitions/Subnet",
+          "description": "The Subnet that using the prefix of this IpAllocation resource."
+        },
+        "virtualNetwork": {
+          "$ref": "./virtualNetwork.json#/definitions/VirtualNetwork",
+          "description": "The VirtualNetwork that using the prefix of this IpAllocation resource."
+        },
+        "type": {
+          "$ref": "#/definitions/IpAllocationType",
+          "description": "The type for the IpAllocation."
+        },
+        "prefix": {
+          "type": "string",
+          "description": "The address prefix for the IpAllocation."
+        },
+        "prefixLength": {
+          "type": "integer",
+          "x-nullable": true,
+          "default": 0,
+          "description": "The address prefix length for the IpAllocation."
+        },
+        "prefixType": {
+          "$ref": "./network.json#/definitions/IPVersion",
+          "default": null,
+          "description": "The address prefix Type for the IpAllocation."
+        },
+        "ipamAllocationId": {
+          "type": "string",
+          "description": "The IPAM allocation ID."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "IpAllocation tags."
+        }
```

Comment:
tags are part of the top-level definition, not inside the property bag

Reply:
This resource have the tag attribute, and the tags property here is by design, this resource will use the properties.tags for internal usage.

Reply:
If you need something like this please call it something other than "tags".  "Tags" means something very specific in ARM

Reply:
We already have NRP code deployed to canary environment and create some resource, also this is a internal feature for hypernet, so is there any we can keep the NRO side code not change and mapping the NRP side tags  to other name?

Reply:
Fixed by change Tags to AllocationTags.